### ISSUE TITLE
Do not fill end time for duration >= 24 hours

### ISSUE THREAD

**Issue by StefanWinterberger** [kimai/kimai#4230]
### Describe the issue

In the form for creating new entries (My Times > Create [in standard mode]):

If I enter a duration that leads to an endtime on the next day or later, the form field for the end time gets filled in correctly, but assume the same date as the start time. When saving this entry, this leads to an unexpected behaviour.

An example: I create an Entry with From Date: 8/9/2023 and Time 11:00 AM. Then I enter a duration of 30 hours.
The End time gets set to 5:00 PM automatically, which would be correct but on the next day (8/10/2023).
When saving this entry, it creates an entry from 8/9/2023 11:00 AM until 8/9/2023 5:00 PM with a duration of only 6 hours.

A possible workaround is to manually delete the end time again, then the entry gets created correctly with a duraiton of 30 hours.

I could reproduce this in the demo instance as well.

Possible solutions I can think of:
- To not set the end time automatically, if a duration is set.
- Or change the handling of the date of the end time in the background.

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

2.0.29

### How do you run Kimai?

Docker

### Which PHP version are you using?

8.1

**Comment by kevinpapst** [comment]
Records longer than 24 hours are not supported via the UI.
If you need it, clear the end field and let Kimai calculate the end time.
YAGNI: Obviously this is super edge case and usually not required. Such a timesheet just doesn't make sense.

But it is an idea to not fill the end time, once the duration exceeeds 24 hours.

**Comment by StefanWinterberger** [comment]
Thank you for your feedback.
I agree that it is an edge case. Our HR people need it to enter records for employees that work externaly and don't have access to our Kimai instance themselves. They want then to make a single entry for the time this employee worked for us per week or month.

**Comment by kevinpapst** [comment]
Let them use the weekly timesheet view. 
That is a couple of clicks more, but it is the correct way to collect this information.

**Comment by StefanWinterberger** [comment]
I will recommend it, and see if it fits their needs.
But they cannot add entries for other employees in the weekly timesheet, right?

**Comment by kevinpapst** [comment]
Ah damn... you are right. Doesn't work for now I guess

**Comment by klein-hirn** [comment]
I'm having a similar issue: I'm tracking working times including "on call" activities. 24 hour shifts are not unusual in some industries (hospitals, healthcare, fire departments...), and sometimes, due to the nature of emergencies, working hours and therefore tracked activities can even exceed 24 hours.

I've found that kimai can handle these durations just fine and also bills them correctly; but there's no way to enter them in the web frontend.

Specifically I found the Kimai mobile app by Cloudrizon GmbH can enter these working hours (it lets the user enter start and end date additionally to time). So I create the entry in web (to adjust hourly rate), and edit in the app for correct end time, et voila, correct summation and billing for all activities even beyond 24 hours (but as soon as I re-edit in kimai web frontend, duration gets reset to <24h). 

Obviously this is a workaround. 

As this issue now got tagged as feature request and this is simply a matter of being able to adjust the end date for an activity, I would request/suggest a simple checkbox next to the end time for "next day" (I know other products work this way, namely SP-EXPERT). This would allow simple tracking of activities exceeding 24h (but not 48); I know of no industry where it is legal to work 48h straight or even close to that, so it would probably satisfy most needs.

Thanks for your consideration and all the awesome work you're putting in this project.

**Comment by kevinpapst** [comment]
I doubt that is legal to work longer than 24 hours at a time. Anywhere.
Create two entries. Done.

I don't think that I am going to pollute the frontend with another form element that is used in 0,001% of all cases (or less).

**Comment by klein-hirn** [comment]
Sorry for getting emotional (bear with me, it gets better); I know it's not something white-colar/desk workers are used to and a lot of people like to forget, but let me assure you 24 hour shifts are still a thing for some "less priviledged" workers (I'm on a 24h shift as emergency physician / Notarzt right now, as a matter of fact; it's my freelance side-job). So please don't discedit people's work reality just because you're lucky enough to never having to do it.

To keep it factual: I can only speak for Germany, but here 24h is legal as standby duty (Bereitschaftsdienst) or on-call duty (Rufbereitschaftsdienst) and standard practice in most smaller hospitals and a most of the large ones (i.e. in my pevious job at a 1200 bed / >8000 employee university hospital). This also applies to the majority of fire departments and emergency services. 24 hours is at the same time the legal maximum, but this can (legally) be exceeded in case of an emergency (take my case today: if we're alerted just prior to the end of my shift tomorrow at 8, we obviously can't wait for the next shift to start and I also can't be relieved "out in the field"; same with a fire department, they won't just not come because the clock says it's almost time to go home, so overtime after a 24h shift does happen and isn't even that rare).

I do however understand that kimai is mostly used by freelancers and IT-workers, so this is a rather special / rare case that might not justify changig the frontend for everyone. I just saw Stefan's issue and thought I'd add my voice to it. 
Maybe this can be shown conditionally when allowed activity duration is >24h (that option does exist in config). Alternatively as a plugin, since it seems to be "just" a frontend modification.

**Comment by kevinpapst** [comment]
> So please don't discedit people's work reality just because you're lucky enough to never having to do it.

Are you serious? When and where did I do that? 
Don't put words in my mouth I neither said nor meant. 

To keep it factual:

1. Remove the end time, then you can enter arbitrary durations
2. According to all sites I read (both for German and Austrian law) even 24 hour shifts require break-times

**Comment by klein-hirn** [comment]
Sorry, didn't mean to step on your toes (well I did, but I obviously overshot; for that I apologize). Yes, breaks are required. In theory. This is touching a nerve with me because it's happened with more than one employer that I worked overtime and just couldn't enter it in time tracking (not kimai), thus not even being paid for it because the system simply didn't allow such "unexpected behavior" to be registered; and what isn't in the system doesn't exist for HR.

Regarding 1: I don't know why, but it took me several tries to get this to work. Thanks, this does solve the issue for me.

**Comment by kevinpapst** [comment]
If anyone else doesn't understand how to enter:

- choose date
- choose start-time
- enter duration
- empty the end field 
- save

This will create a time-entry with the entered duration, the end-time is correctly calculated by Kimai.

**Comment by ruudschilders** [comment]
Hi! I also ran into this.

I am standby for the customer from 18:00-08:00 , which is 14 hours so no problem.
But I am also standby in the weekends, so I used to book a time Friday 18:00 - Monday 08:00 which is 62 hours.
After upgrade to the current version when I do that, I have the same issue.

So I don't know if booking hours for standby is an edge case? Anyway, I am happy I found the workaround suggested by OP

**Comment by kevinpapst** [comment]
Yes, I consider entries > 24h edge cases. Still if the duration is longer than 24 hours the end should not be populated. 

If YOU are affected, please vote on the initial issue with 👍  so it gets higher visibility.
